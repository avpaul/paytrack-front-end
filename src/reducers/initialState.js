import { generateID } from '../actions';

export default {
  accounts: [
    {
      id: generateID(),
      name: 'cash',
      balance: '43000',
    },
    {
      id: generateID(),
      name: 'bk',
      balance: '340200',
    },
    {
      id: generateID(),
      name: 'momo',
      balance: '43500',
    },
  ],
  records: [
    {
      id: generateID(),
      category: 'clothes & shoes',
      account: 'cash',
      notes: 'Bought a new pair of shoes for the coming party',
      amount: 38000,
      date: '05/28/2019',
      type: 'expense',
    },
    {
      id: generateID(),
      category: 'income',
      account: 'bk',
      notes: 'May 2019 salary',
      amount: 285499,
      date: '05/27/2019',
      type: 'income',
    },
    {
      id: generateID(),
      category: 'life & entertainment',
      account: 'cash',
      notes: "Celebrate Patrick's BD",
      amount: 20000,
      date: '05/06/2019',
      type: 'expense',
    },
  ],
};
