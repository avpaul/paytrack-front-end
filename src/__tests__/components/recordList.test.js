import React from "react";
import { shallow } from "enzyme";
import joJson, { toJson } from "enzyme-to-json";
import configureStore from "redux-mock-store";
import RecordList from "../../components/recordList";
import initialState from "../../reducers/initialState";

const mockStore = configureStore();
const store = mockStore(initialState);

describe("render()", () => {
  test("should render the component", () => {
    const wrapper = shallow(<RecordList records={initialState.records} />);
    // const component = wrapper.dive();
    expect(wrapper).toMatchSnapshot();
  });
});
