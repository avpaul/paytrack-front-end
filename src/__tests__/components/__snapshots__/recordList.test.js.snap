// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() should render the component 1`] = `ShallowWrapper {}`;
